### PR TITLE
Pay 5323

### PR DESCRIPTION
**Pull request by priyatham-bhikkumalla**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java [hunk 1]
```diff
@@ -203,6 +197,110 @@ public ResponseEntity updateTheRemissionAmount(String paymentReference, Resubmit
         return new ResponseEntity<>(null, HttpStatus.OK);
     }
 
+    @Override
+    public PaymentGroupResponse checkRefundAgainstRemission(MultiValueMap<String, String> headers,
+                                                            PaymentGroupResponse paymentGroupResponse, String ccdCaseNumber) {
+
```
Comment (kaushikmaitra) [inline api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java:203]: Some log statements would be very helpful along with comments as to what each block of code is doing.

File: api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java [hunk 2]
```diff
@@ -203,6 +197,110 @@ public ResponseEntity updateTheRemissionAmount(String paymentReference, Resubmit
...
+    }
+
+    private RefundListDtoResponse getRefundsFromRefundService(String ccdCaseNumber, MultiValueMap<String, String> headers) {
+
+
+        RefundListDtoResponse refundListDtoResponse;
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(refundApiUrl + REFUND_ENDPOINT).queryParam("ccdCaseNumber",ccdCaseNumber);
+
+        LOG.debug("builder.toUriString() : {}", builder.toUriString());
```
Comment (kaushikmaitra) [inline api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java:263]: You should if log.isDebugEnabled() and if true , then add the log.debug statement

File: api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java (hunk 1 above)
Comment (priyatham-bhikkumalla) [reply api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java]: Added comments and logs

File: api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java (hunk 2 above)
Comment (priyatham-bhikkumalla) [reply api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java]: > You should if log.isDebugEnabled() and if true , then add the log.debug statement

taken care

**Review by ranjeet-kumarhmcts — APPROVED** [review]

**Pull request merged**